packages/react/src/components/ComboButton/ComboButton-test.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright IBM Corp. 2016, 2023
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+
+import { MenuItem } from '../Menu';
+
+import { ComboButton } from './';
+
+describe('ComboButton', () => {
+  it('should support a ref on the outermost element', () => {
+    const ref = jest.fn();
+    const { container } = render(
+      <ComboButton label="Primary action" ref={ref}>
+        <MenuItem label="Additional action" />
+      </ComboButton>
+    );
+    expect(ref).toHaveBeenCalledWith(container.firstChild);
+  });
+
+  it('should support a custom class name on the outermost element', () => {
+    const { container } = render(
+      <ComboButton label="Primary action" className="test">
+        <MenuItem label="Additional action" />
+      </ComboButton>
+    );
+    expect(container.firstChild).toHaveClass('test');
+  });
+
+  it('should forward additional props on the outermost element', () => {
+    const { container } = render(
+      <ComboButton label="Primary action" data-testid="test">
+        <MenuItem label="Additional action" />
+      </ComboButton>
+    );
+    expect(container.firstChild).toHaveAttribute('data-testid', 'test');
+  });
+
+  it('should render props.label on the trigger button', () => {
+    render(
+      <ComboButton label="Test">
+        <MenuItem label="Additional action" />
+      </ComboButton>
+    );
+    expect(screen.getAllByRole('button')[0]).toHaveTextContent(/^Test$/);
+  });
+
+  it('should emit props.onClick on primary action click', async () => {
+    const onClick = jest.fn();
+    render(
+      <ComboButton label="Test" onClick={onClick}>
+        <MenuItem label="Additional action" />
+      </ComboButton>
+    );
+
+    expect(onClick).toHaveBeenCalledTimes(0);
+    await userEvent.click(screen.getAllByRole('button')[0]);
+    expect(onClick).toHaveBeenCalledTimes(1);
+  });
+
+  it('should open a menu on click on the trigger button', async () => {
+    render(
+      <ComboButton label="Primary action">
+        <MenuItem label="Additional action" />
+      </ComboButton>
+    );
+
+    await userEvent.click(screen.getAllByRole('button')[1]);
+
+    expect(screen.getByRole('menu')).toBeTruthy();
+    expect(screen.getByRole('menuitem')).toHaveTextContent(
+      /^Additional action$/
+    );
+  });
+
+  it('should support being disabled', () => {
+    render(
+      <ComboButton label="Primary action" disabled>
+        <MenuItem label="Additional action" />
+      </ComboButton>
+    );
+
+    // primary action button
+    expect(screen.getAllByRole('button')[0]).toBeDisabled();
+
+    // trigger button
+    expect(screen.getAllByRole('button')[1]).toBeDisabled();
+  });
+});
```

Comment:
should this file also verify for the [`children`](https://github.com/carbon-design-system/carbon/pull/13204/files#diff-051197835517702352195445ea71c7fdf1322b984526104996a020a6eb16a87dR42-R56), [`size`](https://github.com/carbon-design-system/carbon/blob/main/packages/react/src/components/DataTable/__tests__/Table-test.js#LL12-L19C5), `tooltipAlign` and [`translateWithId`](https://github.com/carbon-design-system/carbon/pull/13204/files#diff-051197835517702352195445ea71c7fdf1322b984526104996a020a6eb16a87dR149-R169) props? 

A [snapshot test](https://github.com/carbon-design-system/carbon/pull/13204/files#diff-051197835517702352195445ea71c7fdf1322b984526104996a020a6eb16a87dR15-R26) at the top might also be nice

Can we also separate the tests into "'renders as expected - Component API'" and "'behaves as expected'" categories to follow standard? See example file: https://github.com/carbon-design-system/carbon/blob/main/packages/react/src/components/DataTable/__tests__/TableHeader-test.js

Reply:
I added tests for `size`, `tooltipAlign` and `translateWithId`. Not sure if and how we wanna test `children` as it should be taken care of by `Menu` I think.

irt the snapshot: I'm usually not the biggest fan of snapshotting the DOM but if you'd like I can add it as well. Wasn't sure what Carbon's standpoint is here as it's not consistent throughout the codebase.

Reply:
`children` is usually tested in a simple way to ensure composition works, like this

https://github.com/carbon-design-system/carbon/blob/36685a0be5e62db11a60d938c76b1c49a402c99a/packages/react/src/components/Button/__tests__/Button-test.js#L14-L21

and yeah I'd also prefer to have specific tests with proper assertions instead of using `.toMatchSnapshot` where possible

Reply:
I think the `opens a menu on click` test should cover that, correct?

https://github.com/janhassel/carbon/blob/87b683d3ae01032c08d4766b4a0ef9a4d9e8be0b/packages/react/src/components/ComboButton/ComboButton-test.js#L147-L160

Reply:
you're right! I think that should be good enough 🚀